src/main/java/com/faforever/client/chat/ChatNavigation.java

```diff
@@ -0,0 +1,41 @@
+package com.faforever.client.chat;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.util.LinkedHashSet;
+
+@Component
+@Lazy
+@Slf4j
+public class ChatNavigation { // TODO: Not a very good name of class
+
+  @Getter
+  private final LinkedHashSet<ChatTab> currentTabs = new LinkedHashSet<>();
+
+  public boolean addTabIfMissing(String tabId) {
+    return currentTabs.add(new ChatTab(tabId));
+  }
+
+  public void removeTab(String tabId) {
+    currentTabs.removeIf(chatTab -> chatTab.getId().equals(tabId));
+  }
+
+  public String getLastOpenedTabId() {
+    return currentTabs.stream().filter(ChatTab::isSelected).findFirst().map(ChatTab::getId).orElse(null);
+  }
+
+  public void setLastOpenedTabId(String tabId) {
+    if (ChatController.ADD_CHANNEL_TAB_ID.equals(tabId)) {
+      return;
+    }
+
+    currentTabs.stream()
+               .peek(chatTab -> chatTab.setSelected(false))
+               .filter(chatTab -> chatTab.getId().equals(tabId))
```

Comment:
We can't have a side effect in peek and use filter with find first because the peek operation is not guaranteed to be used run for every element of the stream.